src/Mutator/Removal/ArrayItemRemoval.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017-2019, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Mutator\Removal;
+
+use Generator;
+use Infection\Config\Exception\InvalidConfigException;
+use Infection\Mutator\Util\Mutator;
+use Infection\Mutator\Util\MutatorConfig;
+use PhpParser\Node;
+
+/**
+ * @internal
+ */
+final class ArrayItemRemoval extends Mutator
+{
+    private const DEFAULT_SETTINGS = [
+        'remove' => 'first',
+        'limit' => PHP_INT_MAX,
+    ];
+
+    private $remove;
+    private $limit;
+
+    public function __construct(MutatorConfig $config)
+    {
+        parent::__construct($config);
+
+        $settings = $this->getResultSettings();
+
+        $this->remove = $settings['remove'];
+        $this->limit = $settings['limit'];
+    }
+
+    public function mutate(Node $arrayNode): Generator
+    {
+        if ($arrayNode instanceof Node\Expr\Array_) {
```

Comment:
Is it needed? `mutatesNode` already checks `$node instanceof Node\Expr\Array_`

Reply:
This is to satisfy Scrutinizer.

Previously I had type annotation there so that code is property autosugggested for `$arrayNode`.
But that caused Scrutinizer to raise a flag that it is possible to pass Node into the method but I used it only in the context of the array.

Reply:
You can add an annotation like:

```php
/**
 * @param Node|Node\Expr\Array_  $arrayNode
 */
```

Which should satisfy most checkers.

Reply:
fixed